src/data_handler/data_handler.cpp

```diff
@@ -1129,4 +1115,68 @@ int get_avg_band_powers (double *raw_data, int rows, int cols, int sampling_rate
     delete[] bands;
 
     return (int)BrainFlowExitCodes::STATUS_OK;
+}
+
+///////////////////////
+/////// Helpers ///////
+///////////////////////
+
+void calc_per_channel_band_powers (double *raw_data, int row_num_start, int row_num_stop, int cols,
+    int nfft, int sampling_rate, int aply_filters, double **bands, int *exit_codes)
+{
+    for (int i = row_num_start; i < row_num_stop; i++)
+    {
+        double *ampls = new double[nfft / 2 + 1];
+        double *freqs = new double[nfft / 2 + 1];
+        double *thread_data = new double[cols];
+        memcpy (thread_data, raw_data + i * cols, sizeof (double) * cols);
+
+        if (aply_filters)
+        {
```

Comment:
Might want to add a log method to state which method failed for the exit code as I don't think the log methods in those methods specify their methods. Not sure if it would be overly clunky though, so it's your call.